tests/conftest.py

```
@@ -1441,3 +1577,55 @@ def clean_migration_home(
+    outcome = yield
+    report = outcome.get_result()
+    if report.when == "call" and report.failed:
+        open_conns = _conn_provenance.report_open()
+        if open_conns:
+            stacks = []
+            for entry in open_conns:
+                stacks.append(
+                    f"  LEAKED conn={entry['conn_id']} db={entry['db_path']} "
+                    f"created_in={entry['test_nodeid']} thread={entry['thread_name']}\n"
+                    f"{entry['creation_stack']}"
+                )
```

Ruff (0.15.10)

[warning] Use a list comprehension to create a transformed list

(PERF401)